src/ripple/app/hook/impl/applyHook.cpp

```diff
@@ -1739,12 +1739,21 @@ DEFINE_JS_FUNCTION(int64_t, trace, JSValue msg, JSValue data, JSValue as_hex)
         JSValue sdata = JS_JSONStringify(ctx, data, replacer, JS_UNDEFINED);
         JS_FreeValue(ctx, replacer);
 
-        size_t len;
-        const char* cstr = JS_ToCStringLen(ctx, &len, sdata);
-        if (len > 1023)
-            len = 1023;
-        out += std::string(cstr, len);
-        JS_FreeCString(ctx, cstr);
+        if (!(JS_IsException(sdata) || JS_IsUndefined(sdata)))
+        {
+            assert(JS_IsString(sdata));
+            size_t len;
+            const char* cstr = JS_ToCStringLen(ctx, &len, sdata);
+            if (len > 1023)
+                len = 1023;
+            out += std::string(cstr, len);
+            JS_FreeCString(ctx, cstr);
+            JS_FreeValue(ctx, sdata);
+        }
+        else
+        {
+            // No need to free value types
```

Comment:
Maybe want to handle this error better .... 
`out += "<failed-to-stringify>"` or some such